aspnetcore/security/authentication/azure-ad-b2c.md

```diff
@@ -4,150 +4,124 @@ author: camsoper
+    1. Add the following `using` directives:
+    
+        :::code language="csharp" source="azure-ad-b2c/sample/Startup.cs" id="snippet_NewUsings":::
 
-## Configure the underlying OpenIdConnectOptions/JwtBearer/Cookie options
+        The preceding code resolves references used in the next steps.
 
-To configure the underlying options directly, use the appropriate scheme constant in `Startup.ConfigureServices`:
+    1. Replace the `ConfigureServices()` method with the following code:
+        
+        :::code language="csharp" source="azure-ad-b2c/sample/Startup.cs" id="snippet_ConfigureServices":::
 
-```csharp
-services.Configure<OpenIdConnectOptions>(
-    AzureAD[B2C]Defaults.OpenIdScheme, options => 
-    {
-        // Omitted for brevity
-    });
+        In the preceding code:
 
-services.Configure<CookieAuthenticationOptions>(
-    AzureAD[B2C]Defaults.CookieScheme, options => 
-    {
-        // Omitted for brevity
-    });
+        - The calls to the `AddAuthentication` and `AddMicrosoftIdentityWebApp` methods configure the app to use Open ID Connect, specifically configured for the Microsoft Identity platform.
+        - The call to the `AddAuthorization` method initializes ASP.NET Core authorization.
+        - The `AddRazorPages` method call is configured such that anonymous browsers can view the Index page. All other requests require authentication.
+        - The `AddMvcOptions` and `AddMicrosoftIdentityUI` method calls add the required UI components for redirecting to/from Azure AD B2C.
+    
+    1. Add the highlighted line to the `Configure` method:
+        
+        :::code language="csharp" source="azure-ad-b2c/sample/Startup.cs" id="snippet_Configure" highlight="19":::
 
-services.Configure<JwtBearerOptions>(
-    AzureAD[B2C]Defaults.JwtBearerAuthenticationScheme, options => 
-    {
-        // Omitted for brevity
-    });
-```
+        The preceding code enables authentication in ASP.NET Core.
 
 ## Run the app
 
-In Visual Studio, press **F5** to build and run the app. After the web app launches, select **Accept** to accept the use of cookies (if prompted), and then select **Sign in**.
-
-![Sign into the app](./azure-ad-b2c/_static/signin.png)
-
-The browser redirects to the Azure AD B2C tenant. Sign in with an existing account (if one was created testing the policies) or select **Sign up now** to create a new account. The **Forgot your password?** link is used to reset a forgotten password.
-
-![Azure AD B2C login](./azure-ad-b2c/_static/b2csts.png)
+1. Run the app.
+    
+    ```dotnetcli
+    dotnet run
+    ```
 
-After successfully signing in, the browser redirects to the web app.
+1. Browse to your app's secure endpoint (usually `https://localhost:5001/`).
+    - The Index page renders with no authentication challenge.
+    - The header includes a "Sign in" link because you're not authenticated.
```

Comment:
```suggestion
1. Browse to the app's secure endpoint, for example, `https://localhost:5001/`.
    - The Index page renders with no authentication challenge.
    - The header includes a **Sign in** link because you're not authenticated.
```